### PR TITLE
[crypto][kimchi] update proof-systems to latest

### PR DESCRIPTION
**Pull request by mimoo**
as the title says

### CODE REVIEW

**Review by bkase — APPROVED** [review]

File: src/lib/crypto/rust-toolchain.toml [hunk 1]
```diff
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "1.56.0"
+channel = "1.58.0"
```
Comment (mrmr1993) [inline src/lib/crypto/rust-toolchain.toml:2]: Every time you do this, it disrupts everyone building `develop`, and they have to re-figure-out how to fix it. As a courtesy, you should update `README-dev` and send a message to eng internal telling people how to fix it.

**Review by mrmr1993 — APPROVED** [review]

File: src/lib/crypto/rust-toolchain.toml (hunk 1 above)
Comment (mimoo) [reply src/lib/crypto/rust-toolchain.toml:2]: README-dev was already updated, this file should have been updated as well.

> it disrupts everyone building develop, and they have to re-figure-out how to fix it

If this file was updated it should work flawlessly

Comment (mimoo) [reply src/lib/crypto/rust-toolchain.toml:2]: Ah, I discovered the issue. There's a `rust-toolchain.toml` at the root, and there's on in `src/lib/crypto`. I think we can remove the one in crypto.

Comment (mimoo) [reply src/lib/crypto/rust-toolchain.toml:2]: although, I'm wondering if dune will build this correctly if it doesn't move that to the _build folder or something...

**Pull request merged**